specs/sharding/beacon-chain.md

```diff
@@ -217,36 +271,110 @@ class ShardBlobBodySummary(Container):
     data_root: Root
     # Latest block root of the Beacon Chain, before shard_blob.slot
     beacon_block_root: Root
+    # Builder of the data, pays data-fee to proposer
+    builder_index: BuilderIndex
+    # TODO: fee payment amount fields (EIP 1559 like)
+```
+
```

Comment:
BTW don't we want to inline 
`ShardBlobBody` -> `ShardBlob` 
`ShardBlobBodySummary` -> `ShardBlobHeader` 
I don't think they are used somewhere else, so may be it's worth to reduce the number of containers?

Reply:
I would, but collapsing the body is useful to reduce the size of a slashing. The `ShardBlobReference` just has a `body_root: Root` instead of a `body_summary: ShardBlobBodySummary`.

Reply:
Different idea to reduce types; we could try aggregate the builder and validator signatures, and specify one or the other is a 0 in aggregation when the builder_index or proposer_index is max_uint64

Reply:
Yeah, really nice solution 👍 
Curious whether we can make `ShardBlobReference` container 'private' to the `process_shard_proposer_slashing` function?